RELEASES.md

```diff
@@ -0,0 +1,188 @@
+or if you think an issue or PR should be included in a milestone, please reach
+out to the maintainers.
+
+### Support Horizon
+
+Each release will be supported for a period of 2x the release cycle,
+effectively ensuring that at any given time there are two supported versions.
+This means we will accept bug reports and bug fixes up until the end of life
+date. With the current release cycle of 6 months, this means that each release
+will be supported for 1 year.
+
+| Version | Status | Start | End of Life |
+|---------|--------|-------|-------------|
+| TBD | Pre | TBD | TBD + 1 year |
+
+TODO: Figure out release versioning with the mess of pre-existing versions from
+older docker releases and then docker-ce releases.
+
+### Backporting
+
+Backports are community driven. As maintainers, we will try to make sure that
+critical bug fixes make it into all active releases, but generally once a merge
+window has opened our focus will be on shaping up the next release.
+
+If there are fixes that need to be backported, please let us know ine one of
+three ways:
+
+1. Open an issue.
+2. Open a PR with the cherry-picked change from master
+3. Open a PR with a ported fix
+
+Any supported release can accept backports. If you are cherry-picking a commit,
+please use `git cherry-pick -x -s <commit>` to ensure the original commit is
+referenced in the commit message and the cherry-pick has a DCO sign-off in
+addition to the original commit.
+
+Please fix the bug in master before submitting a backport.
+
+### Public API stability
+
+| Component | Status | Stablized Version |
+|-----------|--------|-------------------|
+| HTTP API  | Stable | 1.0 |
+| Prometheus Metrics | Unstable | _future_ |
+| Go Client API | Unstable | _future_ |
+
+#### HTTP API
+
+The primary product of Moby is the HTTP API. The HTTP API will not have any
+backwards incompatible changes.
+Any breaking changes to the HTTP API are considered bugs and should be fixed in
+a patch release.
+
+#### Prometheus Metrics
+
+`dockerd` emits Prometheus-style metrics which relate to the metrics of the
+daemon itself (nothing to do with container metrics). These are not considered
+stable.
+
+#### Go Client API
+
+While we generally try not to make breaking changes to the Go client, we do not
+consider the Go client API stable.
+Breaking changes to this API should be detectable at compile time, and it is
+recommended to vendor in the specific versions you need.
+
+
+#### Deprecations
+
+Individual subsystems may implement things based on some value, for example a
+generic driver option, or even a driver itself. These must remain stable, but
+things may be deprecated. Deprecations only signal to users that they should
+migrate off of some feature because it will be removed in the future.
+
+Deprecations must be communicated in [DEPRECATED.md](DEPRECATED.md)
+Deprecated features must not be removed for a period of at least 3 release cycles.
```

Comment:
This may be one to discuss; IIRC, the "3 release cycles" was done at a time where we did (bi?)monthly releases https://github.com/moby/moby/pull/24572, but never revisited when we switched to 3-month cycles (edge/stable), and to 6-month cycles.

(wondering if having a 18-month deprecation cycle will draw us in a corner with the changes to move more things to containerd)

Reply:
I think we've always had 3-month release cycles, just that we changed from 3-month support to 4 months?

Definitely understand wanting to have the flexibility to remove things more quickly.  This is mainly added as a stability guarantee.
Remember that each deprecation is work that we are creating for people using dockerd. Having a long enough period to be able to accomplish this work makes it easier for people to both plan for the removal and upgrade more quickly.

I can see moving this to two cycles considering that I am proposing support for 2x the release cycle (so 1 year), which effectively gives someone 18 months to migrate off of whatever feature.
Does that work for you?

Reply:
Updated to 2 cycles.